File: C# part 2/Arrays/07.SelectionSort/SelectionSort.cs
/*
 Problem 7. Selection sort

    Sorting an array means to arrange its elements in increasing order. Write a program to sort an array.
    Use the Selection sort algorithm: Find the smallest element, move it at the first position, find the smallest from the rest, move it at the second position, etc.

 */
namespace _07.SelectionSort
{
    using System;
    class SelectionSort
    {
        static void Main()
        {
            Console.Write("Enter array length: ");
            int length = int.Parse(Console.ReadLine());

            int[] numbers = new int[length];

            Console.WriteLine("Enter integer values");
            for (int i = 0; i < numbers.Length; i++)
            {
                Console.Write("Array[{0}] = ", i);
                string input = Console.ReadLine();
                int number;
                bool isNumber = int.TryParse(input, out number);
                if (isNumber)
                {
                    numbers[i] = number;
                }
                else
                {
                    Console.WriteLine("Input string was not in a correct format");
                }
            }

            int currentNumber = 0;
            int minNumber = int.MaxValue;
            int position = 0;
            for (int i = 0; i < numbers.Length; i++)
            {
                for (int j = i; j < numbers.Length; j++)
                {
                    currentNumber = numbers[j];
                    if (currentNumber < minNumber)
                    {
                        minNumber = currentNumber;
                        position = j;
                    }
                }
                numbers[position] = numbers[i];
                numbers[i] = minNumber;
                minNumber = int.MaxValue;
            }
            Console.WriteLine("Sort array");
            for (int i = 0; i < numbers.Length; i++)
            {
                Console.Write(numbers[i] + " ");
            }
            Console.WriteLine();
        }
    }
}

File: JavaScript Fundamentals/Javascript Arrays/scripts/selectionSort.js
jsConsole.writeLine('--------------------------------------------------------------------');
jsConsole.writeLine('Problem 5: Sorting an array means to arrange its elements in increasing order.Write a script to sort an array.Use the selection sort algorithm: Find the smallest element, move it at the first position, find the smallest from the rest, move it at the second position, etc.')
jsConsole.writeLine('--------------------------------------------------------------------');
console.log('--------------------------------------------------------------------');
console.log('Problem 5: Sorting an array means to arrange its elements in increasing order.Write a script to sort an array.Use the selection sort algorithm: Find the smallest element, move it at the first position, find the smallest from the rest, move it at the second position, etc.')
console.log('--------------------------------------------------------------------');

console.log('Before: [9, 7, 77, 5, 5, 3, 2, 1, -4, 6, 8, 10]');
jsConsole.writeLine('Before: [9, 7, 77, 5, 5, 3, 2, 1, -4, 6, 8, 10]');

var numbers = [9, 7, 77, 5, 5, 3, 2, 1, -4, 6, 8, 10];
var currentSmallestNumber;
var smallestNumber = numbers[0];
var position;
var index = 0;

for (var j = 0; j < numbers.length; j++) {
    for (var i = index; i < numbers.length; i++) {

        currentSmallestNumber = numbers[i];

        if (smallestNumber >= currentSmallestNumber) {
            smallestNumber = currentSmallestNumber;
            position = i;
        }
    }
    if (position != j) {
        numbers[position] = numbers[j];
        numbers[j] = smallestNumber;
    }
    index++;
    smallestNumber = numbers[index];
}

console.log('After: ' + numbers.join(', '));
jsConsole.writeLine('After: ' + numbers.join(", "));